src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/ImportOrganizationUserCommand.cs

```diff
@@ -0,0 +1,305 @@
+        }
+
+        // Marry existing users
+        var existingUsersEmailsDict = importUserData.ExistingUsers
+            .Where(u => string.IsNullOrWhiteSpace(u.ExternalId))
+            .ToDictionary(u => u.Email);
+        var newUsersEmailsDict = newUsers.ToDictionary(u => u.Email);
+        var newAndExistingUsersIntersection = existingUsersEmailsDict.Keys.Intersect(newUsersEmailsDict.Keys).ToList();
+        var organizationUsers = (await _organizationUserRepository.GetManyAsync(importUserData.ExistingUsers.Select(u => u.Id).ToList())).ToDictionary(u => u.Id);
+        var usersToUpsert = new List<OrganizationUser>();
+
+        foreach (var user in newAndExistingUsersIntersection)
+        {
+            existingUsersEmailsDict.TryGetValue(user, out var existingUser);
+            organizationUsers.TryGetValue(existingUser.Id, out var organizationUser);
+
+            if (organizationUser != null)
+            {
+                organizationUser.ExternalId = newUsersEmailsDict[user].ExternalId;
+                usersToUpsert.Add(organizationUser);
+                importUserData.ExistingExternalUsersIdDict.Add(organizationUser.ExternalId, organizationUser.Id);
+            }
+        }
+        await _organizationUserRepository.UpsertManyAsync(usersToUpsert);
+    }
+
+    private async Task AddNewUsers(Organization organization,
+            IEnumerable<ImportedOrganizationUser> newUsers,
+            OrganizationUserImportData importUserData)
+    {
+        var userInvites = new List<OrganizationUserInviteCommandModel>();
+
+        foreach (var user in newUsers)
+        {
+            userInvites.Add(new OrganizationUserInviteCommandModel(user.Email, user.ExternalId));
+        }
+
+        var commandResult = await InviteUsersAsync(userInvites, organization);
+
+        switch (commandResult)
+        {
+            case Success<InviteOrganizationUsersResponse> result:
+                foreach (var u in result.Value.InvitedUsers)
+                {
+                    importUserData.ExistingExternalUsersIdDict.Add(u.ExternalId, u.Id);
+                }
+                break;
+            case Failure<InviteOrganizationUsersResponse> failure:
+                throw new BadRequestException(failure.Error.Message);
+            default:
+                throw new InvalidOperationException($"Unhandled commandResult type: {commandResult.GetType().Name}");
+        }
+    }
+
+    private async Task<CommandResult<InviteOrganizationUsersResponse>> InviteUsersAsync(List<OrganizationUserInviteCommandModel> invites, Organization organization)
+    {
+        var plan = await _pricingClient.GetPlanOrThrow(organization.PlanType);
+        var inviteOrganization = new InviteOrganization(organization, plan);
+        var request = new InviteOrganizationUsersRequest(invites.ToArray(), inviteOrganization, Guid.Empty, DateTimeOffset.UtcNow);
+
+        return await _inviteOrganizationUsersCommand.InviteImportedOrganizationUsersAsync(request);
```

Comment:
@jrmccannon do we want to use this yet? The invite users command is itself behind a feature flag that hasn't been turned on yet, and I think you've still got a bit of work to do on it. I'm leaning towards using the existing invite user flow just so that we don't have stacked feature flags or PRs blocking eachother.